src/openrct2/world/Footpath.cpp

```diff
@@ -119,15 +119,15 @@ static bool entrance_has_direction(const EntranceElement& entranceElement, int32
     return entranceElement.GetDirections() & (1 << (direction & 3));
 }
 
-TileElement* MapGetFootpathElement(const CoordsXYZ& coords)
+PathElement* MapGetFootpathElement(const CoordsXYZ& coords)
 {
     TileElement* tileElement = MapGetFirstElementAt(coords);
     do
     {
         if (tileElement == nullptr)
             break;
         if (tileElement->GetType() == TileElementType::Path && tileElement->GetBaseZ() == coords.z)
-            return tileElement;
+            return tileElement->AsPath();
```

Comment:
if we moved this to L129 like then it removes the double check on type (i know its just a micro optimisation and compilers would probably do this anyway but personally i prefer this style):
```cpp
auto* pathElement = tileElement->AsPath();
if (pathElement != nullptr && pathElement->GetBaseZ() == coords.z)
    return pathElement;
```

Reply:
Makes sense, applied.